src/chocolatey.resources/helpers/functions/Get-ChocolateyConfigValue.ps1

```diff
@@ -0,0 +1,65 @@
+﻿# Copyright © 2022 Chocolatey Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Get-ChocolateyConfigValue {
+    <#
+.SYNOPSIS
+Retrieve a value from the Chocolatey Configuration file
+
+.DESCRIPTION
+This function will attempt to retrieve the path according to the specified Path Type
+to a valid location that can be used by maintainers in certain scenarios.
+
+.NOTES
+Available in 2.1.0+
+
+.INPUTS
+None
+
+.OUTPUTS
+This function outputs the value of the specified configuration key.
+If the key is not found, there is no output.
+
+.PARAMETER configKey
+The name of the configuration value that should be looked up.
+
+.PARAMETER IgnoredArguments
+Allows splatting with arguments that do not apply. Do not use directly.
+
+.EXAMPLE
+>
+$value = Get-ChocolateyConfigValue -configKey 'cacheLocation'
+#>
+    param(
+        [parameter(Mandatory = $true)]
+        [string] $configKey,
+        [parameter(ValueFromRemainingArguments = $true)]
+        [Object[]] $ignoredArguments
+    )
+
+    try {
+        $installLocation = Get-ChocolateyPath -pathType 'InstallPath'
+        $configPath = Join-Path $installLocation "config\chocolatey.config"
+        [xml]$configContents = Get-Content -Path $configPath
+        return $configContents.chocolatey.config.add | `
+                Where-Object { $_.key -eq $configKey } | `
+                Select-Object -ExpandProperty value
+    }
+    catch {
+        Write-Warning "Unable to read config value '$configKey' with error"
+        Write-Warning $_
+        return ""
```

Comment:
The Comment Based Help on Lines 31-33 says that we will output nothing if we can't read the key. As such, we should probably just not have a return here.

```suggestion
```

This would also keep with PowerShell expectations where a Get that can't find what it's looking for either throws an error or returns nothing.

Reply:
In fact... Do we know what scenario we might expect an error here? It would be good to understand how to force this code path, but I'm not entirely sure short of a corrupted config file which Chocolatey may already correct or error beforehand (unless maybe non-admin 🤔 )

Reply:
I think it would take either corrupted xml syntax or a missing/no read permissions config file to trigger this path.

Reply:
Removed the return statement

Reply:
I've discussed with the team this bit. I'm going to update this to Write-Error with the exception object attached so that the package author can handle it in a PowerShell way instead of just swallowing the exception. I have been able to cause this catch to be triggered, but only in wholly unsupported ways.